lib/lambda_ethereum_consensus/p2p/gossip/attestation.ex

```diff
@@ -0,0 +1,87 @@
+defmodule LambdaEthereumConsensus.P2P.Gossip.Attestation do
+  @moduledoc """
+  This module handles attestation gossipsub topics.
+  """
+  use GenServer
+
+  alias LambdaEthereumConsensus.Beacon.BeaconChain
+  alias LambdaEthereumConsensus.Libp2pPort
+  alias LambdaEthereumConsensus.P2P
+
+  @subnet_id_start byte_size("/eth2/00000000/beacon_attestation_")
+  @subnet_id_end byte_size("/ssz_snappy")
+
+  def start_link(init_arg) do
+    GenServer.start_link(__MODULE__, init_arg, name: __MODULE__)
+  end
+
+  def join(subnet_id) do
+    topic = get_topic_name(subnet_id)
+    # TODO: update ENR
+    Libp2pPort.join_topic(topic)
+    P2P.Metadata.set_attestation_subnet(subnet_id, true)
+  end
+
+  def leave(subnet_id) do
+    topic = get_topic_name(subnet_id)
+    # TODO: update ENR
+    Libp2pPort.leave_topic(topic)
+    P2P.Metadata.set_attestation_subnet(subnet_id, false)
+  end
+
+  def collect(subnet_id), do: GenServer.call(__MODULE__, {:collect, subnet_id})
```

Comment:
who calls these?

Reply:
No-one yet. I added these while working on subscriptions, but turns out it wasn't needed. We'll need them when implementing the aggregation of attestations, however.

Reply:
I removed them